### PR TITLE
Feature/provider consumer example

### PR DESCRIPTION
**Pull request by sebplorenz**
Hi, I've added a provider-consumer example. Hope this helps.
Regards, Sebastian

### CODE REVIEW

File: dataspace-connector/provider-consumer/README.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+General Description
+
+### Introduction
+This is a simple producer-consumer-example. In this example you will
+- start up two DataspaceConnectors
+- register a resource at the provider connector using a shell script
+- negotiate a contract at the consumer connector
+- consume the resource
+
+For the sake of simplicity this example uses the DataspaceConnector with an in-built H2 database.
```
Comment (brianjahnke) [inline dataspace-connector/provider-consumer/README.md:14]: Hi,
it is Dataspace Connector.

**Review by brianjahnke — CHANGES_REQUESTED** [review]

File: dataspace-connector/provider-consumer/config_provider.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  "ids:trustStore" : {
+    "@id" : "file:///conf/truststore.p12"
+  },
+  "ids:connectorStatus" : {
+    "@id" : "idsc:CONNECTOR_OFFLINE"
+  },
+  "ids:keyStore" : {
+    "@id" : "file:///conf/keystore-localhost.p12"
+  }
+}
```
Comment (brianjahnke) [inline dataspace-connector/provider-consumer/config_provider.json:62]: missing eof

**Review by brianjahnke — CHANGES_REQUESTED** [review]

File: dataspace-connector/provider-consumer/README.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+| artifactIds  | http://provider:8080/api/artifacts/<artifact_id> <br />in my case<br /> http://provider:8080/api/artifacts/c0041c69-9515-4d54-8914-16380b80f236   | The id of the artifact we want to consume. As found in the previous resource description response. |
+| download     | false      |  |
+
+The request body must contain a contract offer. This contract must match the one the resource was created with. 
+The shell script we used to create the resource added a __USE__ permission to the resource. 
+Therefore, we offer a policy with a __USE__ permission.
+```json
+[ {
+        "@type" : "ids:Permission",
+        "@id" : "http://localhost:8080/api/rules/5041c3e5-4933-419d-87d9-474b98ced678",
```
[the remark this answers is not in the document]
Comment (HeinrichPet) [reply dataspace-connector/provider-consumer/README.md:165]: shouldn't that be changed for all our examples?
Like Step 5 here: [slim readme](https://github.com/International-Data-Spaces-Association/IDS-Deployment-Examples/blob/main/dataspace-connector/slim/docker/README.md)
or here: [Full Readme](https://github.com/International-Data-Spaces-Association/IDS-Deployment-Examples/blob/main/dataspace-connector/full/README.md)

[the remark this answers is not in the document]
Comment (brianjahnke) [reply dataspace-connector/provider-consumer/README.md:165]: @HeinrichPet I dont think so, the difference is, that the examples in the other places reference how the user interacts with the docker containers via swagger. This json payload is used for inter container communication.

File: dataspace-connector/provider-consumer/config_provider.json (hunk 2 above)
Comment (sebplorenz) [reply dataspace-connector/provider-consumer/config_provider.json:62]: What do you mean here? Is there anything needed after the keystore? The proxy stuff that is contained in the original configuration is removed.

Comment (brianjahnke) [reply dataspace-connector/provider-consumer/config_provider.json:62]: It is missing an empty line at the end of the file

**Review by brianjahnke — APPROVED** [review]

**Comment by agarciagan** [comment]
Thanks for the example, we found it very useful. . We have created a postman collection automatically generating the calls of the step-by-step example of the readme. Should we create a new PR justo to attach the collection and include a sentence on the readme?

Regards

Ander

**Comment by sebplorenz** [comment]
Sure. I'd like to see it. I assume this would be less error-prone than following the manual steps from the readme.

**Comment by agarciagan** [comment]
I am trying to update the example from the 5.1.2 version of the IDS Connector to the 6.0.0 version from the full example. I manage to get working one container for the provider, consumer, postgress consumer, postgress provider, ui consumer and ui provider.

I update the config of both ids connectos to disable security and remove the proxy information. The create_resource script is working. I see the resource from the ui.

However, i can not access the resource from the consumer, the example REST calls are giving me a "Message handling or processing failed." error when calling to "/api/ids/description" endpoint.

Have you tried to update the example? Could you give me any hint as how to proceed?

Best regarsd

**Comment by sebplorenz** [comment]
Hi, I'm getting the same error when negotiating the contract. The contract negotiation sequence did not change. Therefore, I assume this is a bug and reported it here: https://github.com/International-Data-Spaces-Association/DataspaceConnector/issues/505

**Comment by agarciagan** [comment]
We have managed to solve the error with minor changes on the message body. I will merge the working example first week of september

**Pull request merged**